torchbenchmark/models/gen_hf_generative.py

```diff
@@ -0,0 +1,109 @@
+import os
+from pathlib import Path
+
+class_models = [
+    ('hf_GPT2', (8, 256), 512, 'AutoConfig.from_pretrained("gpt2")', 'AutoModelForCausalLM.from_config'),
+    ('hf_T5', (8, 512), 256, 'AutoConfig.from_pretrained("t5-small", bos_token_id=0)', 'AutoModelForSeq2SeqLM.from_config'),
+    ('hf_Bart', (8, 256), 256, 'AutoConfig.from_pretrained("facebook/bart-base")', 'AutoModelForMaskedLM.from_config'),
+    ('hf_Reformer', (8, 4096), 3072, 'ReformerConfig(is_decoder=True, bos_token_id=0)', 'AutoModelForCausalLM.from_config'),
+    ('hf_BigBird', (2, 1024), 2048, 'BigBirdConfig(attention_type="block_sparse",)', 'AutoModelForCausalLM.from_config'),
+    ('hf_Albert', (8, 512), 256, 'AutoConfig.from_pretrained("albert-base-v2")', 'AutoModelForMaskedLM.from_config'),
+    ('hf_DistilBert', (8, 512), 256, 'AutoConfig.from_pretrained("distilbert-base-uncased")', 'AutoModelForMaskedLM.from_config')
+]
+for name, input_shape, eval_context, config, model in class_models:
+    folder = Path(name)
+    if not os.path.isdir(folder):
+        os.makedirs(folder)
+    init_program = f"""
+# Generated by gen_hf_generative.py
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import NLP
+from transformers import *
+from datasets import load_dataset
+
+class Model(BenchmarkModel):
+    task = NLP.LANGUAGE_MODELING
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+
+
+        config = {config}
+        self.model = {model}(config).to(device)
+        self.optimizer = optim.Adam(self.model.parameters(), lr=0.001)
+
+        num_tokens = 5
+
+        input_ids = torch.randint(0, config.vocab_size, {input_shape}).to(device)
```

Comment:
i'm curious about your motivation here.  if you're wanting to make the model exercise different codepaths run to run, this would come at the expense of less deterministic benchmarking.  I'm not sure how much impact the input data has on either, but it would have that impact on both.  My gut would be to just fix the inputs.

Reply:
well, I think it's common for us to use `randn` or `randint` for generating data.

I was quite worried about this, but luckily, I don't think any of the models I'm adding have *very* input-dependent runtimes.

The only exception is Reformer, which has a sort embedded within it. However, I think that sorting procedure's runtime isn't very input dependent, luckily.

Reply:
ok. so my objection isn't to using randint, it's to using randint _on every startup_.  Instead, why not have your gen_ script call randint for you and hardcode the tensor in a pkl?  that way at least there wouldn't be run-to-run variation.

Reply:
I think I share @wconstab's concern about a possible jitter, but I was also wondering if we could do some kind of a middle road solution here. We could still generate random inputs, we could just try generating them in a *consistent* way like so w/o affecting the global state.

```python
       # freeze the global rnd state
       with freeze_rng_state():
            torch.manual_seed(42) # set a new state 
            decoder_ids = torch.randint(0, config.vocab_size, {input_shape})
```

Reply:
If we're doing this, we should also make sure the model's parameters are generated consistently as well.

Actually, why don't we just enforce this at the benchmark level? Before you run `__init__`, just reset all of your random seeds.

Reply:
may not be a bad idea, actually. If we do it consistently, for each test.

Reply:
actually.. iirc we do this in the template, and (maybe?) suggest it in the adding models?  should make that more explicit and somehow enforce.  maybe just rely on doing it in the benchmark driver before each test instead?

Reply:
e.g. here:
https://github.com/pytorch/benchmark/blob/master/torchbenchmark/models/Background_Matting/__init__.py#L20

checked, adding_models is out of date. i'll PR to update now. missing some other best practices.. i'll ask you two to weigh in on the PR to get more stuff covered, i'm sure i'll miss something.